src/frontend/js/components/Search/Search.tsx

```diff
@@ -1,37 +1,44 @@
-import * as React from 'react';
+import React from 'react';
 
-import { ResourceListGet } from '../../data/genericSideEffects/getResourceList/actions';
-import { Course } from '../../types/Course';
-import { CourseGlimpseListContainer } from '../CourseGlimpseListContainer/CourseGlimpseListContainer';
+import { useCourseSearch } from '../../data/useCourseSearch/useCourseSearch';
+import {
+  CourseSearchParamsContext,
+  useCourseSearchParams,
+} from '../../data/useCourseSearchParams/useCourseSearchParams';
+import { requestStatus } from '../../types/api';
+import { CourseGlimpseList } from '../CourseGlimpseList/CourseGlimpseList';
 import { SearchFiltersPane } from '../SearchFiltersPane/SearchFiltersPane';
-import { SearchSuggestFieldContainer } from '../SearchSuggestFieldContainer/SearchSuggestFieldContainer';
+import { SearchSuggestField } from '../SearchSuggestField/SearchSuggestField';
 
-export interface SearchProps {
-  requestOrganizations: () => ResourceListGet;
-  requestCategories: () => ResourceListGet;
-}
+export const Search = () => {
+  const [courseSearchParams, setCourseSearchParams] = useCourseSearchParams();
+  const courseSearchResponse = useCourseSearch(courseSearchParams);
 
-interface SearchState {
-  courses: Course[];
-}
-
-export class Search extends React.Component<SearchProps, SearchState> {
-  componentWillMount() {
-    this.props.requestOrganizations();
-    this.props.requestCategories();
-  }
-
-  render() {
-    return (
-      <div className="search">
+  return (
+    <div className="search">
+      <CourseSearchParamsContext.Provider
+        value={[courseSearchParams, setCourseSearchParams]}
+      >
         <div className="search__filters">
-          <SearchFiltersPane />
-        </div>
-        <div className="search__results">
-          <SearchSuggestFieldContainer />
-          <CourseGlimpseListContainer />
+          <SearchFiltersPane
+            filters={
+              courseSearchResponse &&
+              courseSearchResponse.status === requestStatus.SUCCESS
+                ? courseSearchResponse.content.filters
+                : null
+            }
+          />
         </div>
-      </div>
-    );
-  }
-}
+        {courseSearchResponse &&
+        courseSearchResponse.status === requestStatus.SUCCESS ? (
```

Comment:
I'd have to change the children component to accept `false` as well as `null` because `courseSearchResponse.status === requestStatus.SUCCESS` will return `false` not `null` if the status does not match

Reply:
I don't understand, you can write this:

```
{courseSearchResponse &&
          courseSearchResponse.status === requestStatus.SUCCESS && (
            <div className="search__results">
              <SearchSuggestField
                filters={courseSearchResponse.content.filters}
              />
              <CourseGlimpseList
                courses={courseSearchResponse.content.objects}
              />{' '}
            </div>
          )}
```

And you don't have anymore the ternary rendering null

Reply:
Oh right sorry, I thought you meant the one for the `SearchFiltersPane` `filters` prop.